src/randoop/sequence/Sequence.java

```diff
@@ -658,36 +650,49 @@ protected void checkIndex(int i) {
    * Returns a new sequence that is equivalent to this sequence plus the given
    * statement appended at the end.
    */
-  public final Sequence extend(Operation statement, List<Variable> inputVariables) {
-    checkInputs(statement, inputVariables);
+  public final Sequence extend(Operation operation, List<Variable> inputVariables) {
+    checkInputs(operation, inputVariables);
     List<RelativeNegativeIndex> indexList = new ArrayList<RelativeNegativeIndex>(1);
     for (Variable v : inputVariables) {
       indexList.add(getRelativeIndexForVariable(size(), v));
     }
-    Statement newStatement = new Statement(statement, indexList);
-    int newNetSize = (statement instanceof NonreceiverTerm) ? this.savedNetSize : this.savedNetSize + 1;
-    return new Sequence(new OneMoreElementList<Statement>(this.statements, newStatement), this.savedHashCode
-        + newStatement.hashCode(), newNetSize);
+    Statement statement = new Statement(operation, indexList);
+    int newNetSize = (operation.isNonreceivingValue()) ? this.savedNetSize : this.savedNetSize + 1;
+    return new Sequence(new OneMoreElementList<Statement>(this.statements, statement), this.savedHashCode
+        + statement.hashCode(), newNetSize);
   }
 
   /**
    * Returns a new sequence that is equivalent to this sequence plus the given
    * statement appended at the end.
    */
-  public final Sequence extend(Operation statement, Variable... inputs) {
-    return extend(statement, Arrays.asList(inputs));
+  public final Sequence extend(Operation operation, Variable... inputs) {
+    return extend(operation, Arrays.asList(inputs));
   }
 
+  /**
+   * extend adds a new statement to a sequence using the operation of the given statement.
```

Comment:
Better to say "to this sequence" rather than "to a sequence" which led me to look for a parameter to the method.